rust/kernel/sync/arc.rs

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A reference-counted pointer.
+//!
+//! This module implements a way for users to create reference-counted objects and pointers to
+//! them. Such a pointer automatically increments and decrements the count, and drops the
+//! underlying object when it reaches zero.
```

Comment:
Should this mention that these are safely movable between threads?

Reply:
Yes, we should. Reminds me to add `Send` and `Sync` implementations too.

Reply:
And also `Clone`, right? And it's where you increase the underlying  refcount.

Reply:
Ah... miss the following code you `impl Clone`, please ignore me.